### PR TITLE
Change signature v2 to v4

### PR DESCRIPTION
**Pull request by pkoperek**
Hi!

I've noticed that the package in its current form doesn't support V4 signatures. This PR migrates the `alexa_GET` function to use the v4 instead of v2. I based heavily on the `cloudyr/s3` package, so you'll probably notice a lot of similarities.

I've tested it with various calls (url_info, categiry_listing, browse_categories, in_links,etc) and the results looked fine.

Samples:

```
category_listing(path="Top/Arts")
Request ID:  df394c5b-9595-11e8-9cea-5dcb14e9594c 
Response Status:  Success 
# A tibble: 20 x 4
   data_url                popularity_rank title                             description                                                                
   <chr>                   <chr>           <chr>                             <chr>                                                                      
 1 www.youtube.com         2               YouTube                           YouTube is a way to get your videos to the people who matter to you. Uploa…
 2 www.imdb.com            50              The Internet Movie Database       Features plot summaries, reviews, cast lists, and theatre schedules.       
 3 www.bbc.co.uk           83              BBC Online                        The BBC Homepage - Your gateway to BBC Online                              
 4 www.cnn.com             98              CNN                               Cable News Network                                                         
 5 www.espn.com            101             ESPN                              Sports news network. Includes broadcast schedule, game scores and results,…
 6 www.spotify.com         111             Spotify                           Offers a desktop software to listen music by streaming with the possibilit…
 7 www.deviantart.com      135             deviantART                        Gallery of digital works submitted by members.                             
 8 www.hulu.com            178             Hulu                              Ad-supported streaming television shows and movies, including some in high…
 9 www.foxnews.com         181             FoxNews.com                       Breaking News, Latest News and Current News from FoxNews.com. Breaking new…
10 www.livejournal.com     195             LiveJournal.com                   LiveJournal is a rich, community media platform that willfully blurs the l…
11 www.scribd.com          198             Scribd                            A digital documents library that allows users to publish, discover and dis…
12 www.xfinity.com         216             Comcast Corporation               Develops, manages and operates cable communications systems offering regul…
13 www.ign.com             241             IGN Filmforce: Alien Vs. Predator Movie news, previews and behind the scenes information.                    
14 www.rottentomatoes.com  282             Rotten Tomatoes                   Features the consensus opinion of professional critics from across the nat…
15 www.ndtv.com            287             NDTV                              Latest News from India, India News, Business, India Finance, Bollywood, Cr…
16 www.urbandictionary.com 340             Urban Dictionary                  A veritable cornucopia of streetwise lingo, posted and defined by its read…
17 www.discogs.com         358             Discogs                           User built music database. Store and organise personal music collection li…
18 www.ultimate-guitar.com 372             Ultimate Guitar Archive           The largest archive of tabs on the web! Also includes a musicians forum, l…
19 www.cnbc.com            396             CNBC                              Headline news, articles, reports, stocks and quotes, message boards, and a…
20 archiveofourown.org     440             Archive of Our Own                A fan-created archive for transformative fanworks, like fanfiction, fanart…
```

```
browse_categories(path="Top/Arts")
Request ID:  ebdb8d59-9595-11e8-8e54-e5a55655531b 
Response Status:  Success 
# A tibble: 42 x 5
   path                       title             sub_category_count total_listing_count description                                                      
   <chr>                      <chr>             <chr>              <chr>               <chr>                                                            
 1 Top/Arts/Animation         Animation         23                 1573                Sites primarily related to mainstream television and film animat…
 2 Top/Arts/Architecture      Architecture      14                 1197                The art or science of planning and building structures.          
 3 Top/Arts/Art_History       Art History       8                  265                 The Art History category covers all types of art: painting, scul…
 4 Top/Arts/Awards            Awards            3                  7                   General arts and entertainment award sites with multiple award c…
 5 Top/Arts/Bodyart           Bodyart           15                 359                 This category is for decorative body modifications of all kinds,…
 6 Top/Arts/Chats_and_Forums  Chats and Forums  0                  3                   Message boards or discussion forums where one can interact with …
 7 Top/Arts/Classical_Studies Classical Studies 34                 71                  Classics is the study and appreciation of the culture and wisdom…
 8 Top/Arts/Comics            Comics            18                 1318                NA                                                               
 9 Top/Arts/Contests          Contests          0                  0                   "This category, <strong><a href=\"http://www.dmoz.org/Arts/Conte…
10 Top/Arts/Costumes          Costumes          7                  12                  NA                                                               
# ... with 32 more rows
```

I've noticed only one problem `traffic_history` returns an error but it seems to be related to what happens with the result rather than request construction (sorry for Polish error messages ;)):

```
 traffic_history(url = "http://www.google.com", start = "20160505")
Request ID:  047cf0c4-9596-11e8-a0b2-3574c9f05667 
Response Status:  Success 
Błąd w poleceniu 'data.frame(..., check.names = FALSE)':
  arguments imply differing number of rows: 1, 0
```

This is my first R-related PR so please take it lightly :) If you need any changes to be applied please let me know. Also: I tried to do the point 7. from the CONTRIBUTING guide, but failed miserably:

```
$ R CMD BUILD aws.alexa
/usr/lib/R/bin/Rcmd: 62: exec: BUILD: not found
```

any tips how to make this run welcome :)

### CODE REVIEW

**Comment by soodoku** [comment]
cc @leeper

**Comment by soodoku** [comment]
cc @leeper

**Comment by pkoperek** [comment]
btw: I had a quick look at Travis output - I think there is some bigger issue with it not being able to update packages

**Comment by leeper** [comment]
Seems like a good idea.

**Comment by pkoperek** [comment]
what needs to happen to get this integrated?

File: R/aws.alexa.package.R [hunk 1]
```diff
@@ -31,27 +31,70 @@ NULL
...
 #' @return list
 
-alexa_GET <- function(query, key = Sys.getenv("AWS_ACCESS_KEY_ID"),
-                            secret = Sys.getenv("AWS_SECRET_ACCESS_KEY"), ...) {
+alexa_GET <- function(query, 
+                      key = Sys.getenv("AWS_ACCESS_KEY_ID"),
+                      secret = Sys.getenv("AWS_SECRET_ACCESS_KEY"), 
+                      verbose = getOption("verbose", FALSE),
+                      session_token = NULL,
+                      region = 'us-west-1',
```
Comment (leeper) [inline R/aws.alexa.package.R:46]: Why `us-west-1` and not the default `us-east-1`?

Comment (pkoperek) [reply R/aws.alexa.package.R:46]: Not sure if this is related to my account only, but I get #403 in other regions. If you can confirm it works in `us-east-1` I would be more than happy to change this :)

Example:

```
> url_info(url = "google.com", region = 'us-east-1')
BŁĄD: HTTP failure: 403
```

Comment (soodoku) [reply R/aws.alexa.package.R:46]: ok. will merge for now. will investigate this post-merge.

**Pull request merged**